### PR TITLE
docs: adding portal

### PR DESCRIPTION
**Pull request by parsaattari**
Adding Portal to Alchemy smart accounts docs!

### CODE REVIEW

File: site/smart-accounts/signers/portal.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  - - meta
+    - property: og:description
+      content: Guide to use Portal as a signer
+---
+
+# Portal
+
+[Portal](https://www.portalhq.io/) is an embedded blockchain infrastructure company that powers companies with an end to end platform for key management for self-custodial wallets (MPC and AA), security firewall, and web3 protocol connect kit.
+
+A combination of Portal and Account kit allows you to have robust key management and security, while also exploring everything that web3 has to offer with smart contract accounts for your users.
```
Comment (avasisht23) [inline site/smart-accounts/signers/portal.md:19]:
```suggestion
A combination of Portal and Alchemy's Account Kit allows you to have robust key management and security, while also exploring everything that web3 has to offer with smart contract accounts for your users.
```

File: site/smart-accounts/signers/portal.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+## Integration
+
+Check out our developer [docs](https://docs.portalhq.io/) to learn more about Portal. If you want to get quick access to our SDKs, please reach out via [this](https://5g2cefp2j92.typeform.com/portal-labs?typeform-source=www.portalhq.io) form.
+
+### Install the SDK
```
Comment (avasisht23) [inline site/smart-accounts/signers/portal.md:25]:
```suggestion
### Install the Portal SDK
```

File: site/smart-accounts/signers/portal.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+# Portal
+
+[Portal](https://www.portalhq.io/) is an embedded blockchain infrastructure company that powers companies with an end to end platform for key management for self-custodial wallets (MPC and AA), security firewall, and web3 protocol connect kit.
+
+A combination of Portal and Account kit allows you to have robust key management and security, while also exploring everything that web3 has to offer with smart contract accounts for your users.
+
+## Integration
+
+Check out our developer [docs](https://docs.portalhq.io/) to learn more about Portal. If you want to get quick access to our SDKs, please reach out via [this](https://5g2cefp2j92.typeform.com/portal-labs?typeform-source=www.portalhq.io) form.
```
Comment (avasisht23) [inline site/smart-accounts/signers/portal.md:23]:
```suggestion
Check out Portal's developer [docs](https://docs.portalhq.io/) to learn more about Portal. If you want to get quick access to their SDKs, please reach out via [this](https://5g2cefp2j92.typeform.com/portal-labs?typeform-source=www.portalhq.io) form.
```

File: site/smart-accounts/signers/portal.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+```bash [yarn]
+yarn add @portal-hq/web
+```
+
+:::
+
+### Create a SmartAccountSigner
+
+Next, setup the Portal sdk and create a `SmartAccountSigner`:
```
Comment (avasisht23) [inline site/smart-accounts/signers/portal.md:41]:
```suggestion
Next, setup the Portal SDK and create a `SmartAccountSigner`:
```

File: site/snippets/portal.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+} as PortalOptions 
+
+const portal = new Portal(portalOptions)
+
+const portalWalletClient = createWalletClient({
+   transport: custom(portal.provider),
+});
+export const portalSigner: SmartAccountSigner = new WalletClientSigner(
+   portalWalletClient
+);
```
Comment (avasisht23) [inline site/snippets/portal.ts:22]: nit: newline
```suggestion
);

```

File: site/snippets/portal.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import Portal, { PortalOptions } from '@portal-hq/web'
+import { createWalletClient, custom } from "viem";
+import {
+  WalletClientSigner, type SmartAccountSigner 
+} from "@alchemy/aa-core";
+
+
+const portalOptions = {
+  autoApprove: true,
+  gatewayConfig: `https://polygon-mumbai.g.alchemy.com/v2/${process.env.ALCHEMY_API_KEY}`,
```
Comment (avasisht23) [inline site/snippets/portal.ts:10]: nit: can you use viem's chain: https://viem.sh/docs/clients/chains.html and append alchemy api key to it? looks cleaner :)

File: site/smart-accounts/signers/portal.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+```ts [example.ts]
+import { AlchemyProvider } from "@alchemy/aa-alchemy";
+import { LightSmartContractAccount } from "@alchemy/aa-accounts";
+import { polygonMumbai } from "viem/chains";
+import { portalSigner } from "./portalSigner";
+
+const chain = polygonMumbai;
+const provider = new AlchemyProvider({
+  apiKey: "ALCHEMY_API_KEY",
```
Comment (avasisht23) [inline site/smart-accounts/signers/portal.md:58]: nit: for consistency
```suggestion
  apiKey: process.env.ALCHEMY_API_KEY,
```

**Review by avasisht23 — COMMENTED** [review]
nits on wording, styling. 

lgtm otherwise, but deserves a pass from @moldy530 as well, so will wait for when he's back in. otw, will approve tomorrow (wednesday) EOD

File: site/snippets/portal.ts (hunk 6 above)
Comment (avasisht23) [reply site/snippets/portal.ts:10]: bump here

**Comment by avasisht23** [comment]
you'll also need to push a commit with a semantic pr title (e.g. "feat: add portal as a signer option") for our checks to pass

**Comment by avasisht23** [comment]
bump here @parsaattari

File: site/snippets/portal.ts (hunk 6 above)
Comment (parsaattari) [reply site/snippets/portal.ts]: Updated this to `polygonMumbai.rpcUrls.alchemy.http` from `viem/chains`

**Comment by parsaattari** [comment]
@avasisht23 Apologies for the delay in response here. Updated the `viem/chains` recommendation and pushed the last commit with `feat: add portal as a signer option`. Let me know if I need to make any more updates

**Comment by avasisht23** [comment]
hmm @parsaattari it looks like this fails now fails our prettier check? can you take a look? 

https://github.com/OMGWINNING/aa-sdk-staging/actions/runs/6434377949/job/17477321168?pr=65

File: site/.vitepress/config.ts [hunk 8]
```diff
@@ -49,6 +49,7 @@ export default defineConfig({
             base: "/smart-accounts/signers",
             link: "/overview",
             items: [
+              { text: "Portal", link: "/portal" },
               { text: "Capsule", link: "/capsule" },
               { text: "Magic.Link", link: "/magic-link" },
               { text: "Turnkey", link: "/turnkey" },
```
Comment (moldy530) [inline site/.vitepress/config.ts:55]: nit: keeping this alphabetical

```suggestion
              { text: "Capsule", link: "/capsule" },
              { text: "Magic.Link", link: "/magic-link" },
              { text: "Portal", link: "/portal" },
              { text: "Turnkey", link: "/turnkey" },
```

**Pull request merged**